modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,62 @@
+package terraform
+
+import (
+	"errors"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+)
+
+type ValidationOptions struct {
+	RootDir     string
+	ExcludeDirs []string
+}
+
+// readModuleAndExampleSubDirs returns a slice strings representing the filepaths for all valid Terraform modules
+// in both the "modules" directory and "examples" directories in the project root, if they exist.
+func readModuleAndExampleSubDirs(opts ValidationOptions) ([]string, error) {
```

Comment:
This can be a public function. I think this would be useful for other things.

Reply:
Made public in 9519b48